dev-libs/optix/optix-7.3.0.ebuild

```diff
@@ -0,0 +1,124 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake cuda eapi8-dosym
+
+DESCRIPTION="NVIDIA Ray Tracing Engine"
+HOMEPAGE="https://developer.nvidia.com/optix"
+SRC_URI="NVIDIA-OptiX-SDK-${PV}-linux64-x86_64.sh"
+S="${WORKDIR}"
+
+LICENSE="OptiX-SDK"
+SLOT="0/7"
+KEYWORDS="~amd64"
+IUSE="samples"
+
+RESTRICT="fetch mirror"
+
+RDEPEND="
+	dev-util/nvidia-cuda-toolkit
+	media-libs/freeglut
+	virtual/opengl
+	samples? (
+		dev-libs/libbsd
+		media-libs/openexr:0=
+		media-libs/vulkan-loader
+		sys-libs/zlib
+		virtual/opengl
+		x11-libs/libICE
+		x11-libs/libX11
+		x11-libs/libXau
+		x11-libs/libXcursor
+		x11-libs/libXdmcp
+		x11-libs/libXext
+		x11-libs/libXfixes
+		x11-libs/libXinerama
+		x11-libs/libXrandr
+		x11-libs/libXrender
+		x11-libs/libXxf86vm
+		x11-libs/libxcb:=
+	)
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	samples? (
+		app-admin/chrpath
+		dev-libs/libbsd
+		media-libs/openexr:0
+		media-libs/vulkan-loader
+		sys-libs/zlib
+		virtual/opengl
+		x11-libs/libICE
+		x11-libs/libX11
+		x11-libs/libXau
+		x11-libs/libXcursor
+		x11-libs/libXdmcp
+		x11-libs/libXext
+		x11-libs/libXfixes
+		x11-libs/libXinerama
+		x11-libs/libXrandr
+		x11-libs/libXrender
+		x11-libs/libXxf86vm
+		x11-libs/libxcb
+	)
+"
+
+pkg_nofetch() {
+	einfo "Please download ${SRC_URI} from:"
+	einfo "  ${HOMEPAGE}"
+	einfo "and move it to your DISTDIR directory."
+	einfo 'DISTDIR value is available from `emerge --info`'
+}
+
+src_unpack() {
+	tail -n +223 "${DISTDIR}"/${A} | tar -zx || die
+}
+
+src_prepare() {
+	default
+	if use samples; then
+		cuda_src_prepare
+		CMAKE_USE_DIR="${S}/SDK" cmake_src_prepare
+	fi
+}
+
+src_configure() {
+	if use samples; then
+		local mycmakeargs=(
+			# TODO: prepare to use slotted OpenEXR
+#			-DOpenEXR-ROOT="${EPREFIX}"/usr/$(get_libdir)/cmake/OpenEXR
+			-DOpenEXR_INCLUDE_DIR="${EPREFIX}"/usr/include/OpenEXR
+		)
+		CMAKE_USE_DIR="${S}/SDK" cmake_src_configure
+	fi
+}
+
+src_compile() {
+	if use samples; then
+		CMAKE_USE_DIR="${S}/SDK" cmake_src_compile
+	fi
+}
+
+src_install() {
+	insinto /opt/${PN}
+	doins -r include SDK
+	dodoc doc/*
+
+	if use samples; then
+		exeinto /opt/${PN}/bin
+		doexe "${BUILD_DIR}"/bin/*
+		into /opt/${PN}
+		dolib.a "${BUILD_DIR}"/lib/{libDemandLoading.a,libimgui.a}
+		dolib.so "${BUILD_DIR}"/lib/{libglad.so,libsutil_7_sdk.so}
+		chrpath -d "${ED}"/opt/${PN}/bin/* || die
+		chrpath -d "${ED}"/opt/${PN}/$(get_libdir)/*.so* || die
+
+		cat <<- EOF > "${T}"/99${PN}
+		PATH=/opt/${PN}/bin
+		LDPATH=/opt/${PN}/$(get_libdir)
+		EOF
+		doenvd "${T}"/99${PN}
```

Comment:
Although, now that I think about it, this LDPATH use kind of worry me with libglad/libglfw given other packages may use these in the future.

Not really familiar with it but it's some kinda [generator thing](https://github.com/Dav1dde/glad) and I'm not sure these are meant to be shared to the whole system.

Reply:
Is there another way to tell the binaries where to search for needed libraries? I guess, RPATH isn't appropriate here.

Reply:
On that note, I guess you added chrpath because of insecure rpath warning? You can use  `-DCMAKE_SKIP_RPATH=YES` to my cmakeargs to accomplish the same, so chrpath isn't needed here.

But it may not be what we want, it's a bit dodgy given no cmake install, but rpath can be forced at build time:
```
            -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON
            -DCMAKE_INSTALL_RPATH="${EPREFIX}"/path/to/libdir
```

For (at least) things we built, optix* binaries could go in /usr/bin, and libraries in /usr/libdir/${PN} using rpath, and then you wouldn't need this env file at all and private libraries wouldn't pollute.